From 74a24fe1b7e7529f618c32b085177f2be3394c6b Mon Sep 17 00:00:00 2001
From: huang zhibao <hzb@rock-chips.com>
Date: Thu, 27 Aug 2015 16:03:41 +0800
Subject: [PATCH] ARM: dts: rk3288: move the remote-ctrl of PWM into box dts

Thinking about the PWM remote-ctrl is used by box, we should
put this into box dts.

Meanwhile, that PWM configure will break the normal PWM driver.
This patch will fix it.

Change-Id: I14878332631f94f512412ced5ceb8689f8bfbe46
Signed-off-by: huang zhibao <hzb@rock-chips.com>
Signed-off-by: Caesar Wang <wxt@rock-chips.com>
---
 arch/arm/boot/dts/rk3288-box.dts |  9 ++++++++-
 arch/arm/boot/dts/rk3288.dtsi    | 13 -------------
 2 files changed, 8 insertions(+), 14 deletions(-)

diff --git a/arch/arm/boot/dts/rk3288-box.dts b/arch/arm/boot/dts/rk3288-box.dts
index a2dcadd308f3..2001d2178b48 100644
--- a/arch/arm/boot/dts/rk3288-box.dts
+++ b/arch/arm/boot/dts/rk3288-box.dts
@@ -909,9 +909,16 @@
 	regulator-name = "vcc_sd";
 	};
 
-&remotectl {
+/*
+ * Due to not have the software of PWM for remotectrl.
+ * We can _*HACK*_ do that as the following.
+ */
+&pwm0 {
+	compatible = "rockchip,remotectl-pwm";
+	remote_pwm_id = <0>;
 	handle_cpu_id = <1>;
 	status = "okay";
+
 	ir_key1{
 		rockchip,usercode = <0x4040>;
 		rockchip,key_table =
diff --git a/arch/arm/boot/dts/rk3288.dtsi b/arch/arm/boot/dts/rk3288.dtsi
index f4f921895f9d..9c13e3706f5c 100755
--- a/arch/arm/boot/dts/rk3288.dtsi
+++ b/arch/arm/boot/dts/rk3288.dtsi
@@ -919,19 +919,6 @@
 		status = "disabled";
 	};
 
-	remotectl: pwm@ff680000 {
-		compatible = "rockchip,remotectl-pwm";
-		reg = <0xff680000 0x10>;
-		#pwm-cells = <2>;
-		pinctrl-names = "default";
-		pinctrl-0 = <&pwm0_pin>;
-		clocks = <&clk_gates11 11>;
-		clock-names = "pclk_pwm";
-		remote_pwm_id = <0>;
-		interrupts = <GIC_SPI 78 IRQ_TYPE_LEVEL_HIGH>;
-		status = "disabled";
-	};
-
 	dvfs {
 
 		vd_arm: vd_arm {
-- 
2.35.3

